core/src/sendmmsg.rs

```diff
@@ -66,11 +94,66 @@ pub fn send_mmsg(sock: &UdpSocket, packets: &mut [Packet]) -> io::Result<usize>
     Ok(npkts)
 }
 
+#[cfg(not(target_os = "linux"))]
+pub fn multicast(
+    sock: &UdpSocket,
+    packet: &mut Packet,
+    dests: &[&SocketAddr],
+) -> io::Result<usize> {
+    let count = dests.len();
+    for a in dests {
+        sock.send_to(&packet.data[..packet.meta.size], a)?;
+    }
+
+    Ok(count)
+}
+
+#[cfg(target_os = "linux")]
+pub fn multicast(
+    sock: &UdpSocket,
+    packet: &mut Packet,
+    dests: &[&SocketAddr],
+) -> io::Result<usize> {
+    use libc::{sendmmsg, socklen_t};
+    use std::os::unix::io::AsRawFd;
+
+    let mut iovs: Vec<iovec> = Vec::with_capacity(dests.len());
```

Comment:
I'd be nice to have a comment here about why we needed to pre-alloc. It was quite a subtle bug.